tests/test_core/test_core.py

```diff
@@ -201,3 +208,39 @@ def test_opt_in_subcommand(
     # Check the contents of the config file match
     with open(config_file_path) as f:
         assert dict(yaml.safe_load(f)) == expected_configuration
+
+
+def test_unlock_state_lock_called(runner, mock_unlock):
+    """Test if unlock function in RemoteStateManager is called.
+
+    Args:
+        runner: Mock runner
+        mock_unlock: pytest fixture to force unlock state file
+    """
+
+    # Invoke force-unlock command with yes confirmation
+    result = runner.invoke(app, ["force-unlock"], input="Y\n")
+
+    # Exit code 0 means there was no error
+    assert result.exit_code == 0
+
+    # Check if unlock function from RemoteStateManager class is called
+    mock_unlock.assert_called_once()
+
+
+def test_unlock_state_lock_not_called(runner, mock_unlock):
+    """Test if unlock function in RemoteStateManager is not called.
```

Comment:
`"""Test if unlock function in RemoteStateManager is not called when running the user does not confirm.`

Reply:
My mistake this doesn't make sense `"""Test that the unlock function is not called when the user does not confirm.`, does this sound better?